In [1]:
import numpy as np
import pandas as pd
import os

# Flowers

### Preprocessing

In [2]:
iris=pd.read_csv('iris.csv')

In [3]:
iris

sepal.length  sepal.width  petal.length  petal.width    variety
0             5.1          3.5           1.4          0.2     Setosa
1             4.9          3.0           1.4          0.2     Setosa
2             4.7          3.2           1.3          0.2     Setosa
3             4.6          3.1           1.5          0.2     Setosa
4             5.0          3.6           1.4          0.2     Setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  Virginica
146           6.3          2.5           5.0          1.9  Virginica
147           6.5          3.0           5.2          2.0  Virginica
148           6.2          3.4           5.4          2.3  Virginica
149           5.9          3.0           5.1          1.8  Virginica

[150 rows x 5 columns]

Loaded the dataset.

In [4]:
iris.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal.length  150 non-null    float64
 1   sepal.width   150 non-null    float64
 2   petal.length  150 non-null    float64
 3   petal.width   150 non-null    float64
 4   variety       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


We can see there are no null values.

In [5]:
iris[iris.duplicated()==True]

sepal.length  sepal.width  petal.length  petal.width    variety
142           5.8          2.7           5.1          1.9  Virginica

In [6]:
iris=iris.drop_duplicates()

The dataset had a duplicate row which had to be removed.

### Tasks

**Task 1**

In [7]:
iris.variety.value_counts()

Versicolor    50
Setosa        50
Virginica     49
Name: variety, dtype: int64

We can see the count of each type of flower. The data is evenly distributed.

**Task 2**

In [8]:
iris_virginica=iris[iris.variety=='Virginica']

In [9]:
iris_virginica[iris_virginica['petal.length']>1.5]

sepal.length  sepal.width  petal.length  petal.width    variety
100           6.3          3.3           6.0          2.5  Virginica
101           5.8          2.7           5.1          1.9  Virginica
102           7.1          3.0           5.9          2.1  Virginica
103           6.3          2.9           5.6          1.8  Virginica
104           6.5          3.0           5.8          2.2  Virginica
105           7.6          3.0           6.6          2.1  Virginica
106           4.9          2.5           4.5          1.7  Virginica
107           7.3          2.9           6.3          1.8  Virginica
108           6.7          2.5           5.8          1.8  Virginica
109           7.2          3.6           6.1          2.5  Virginica
110           6.5          3.2           5.1          2.0  Virginica
111           6.4          2.7           5.3          1.9  Virginica
112           6.8          3.0           5.5          2.1  Virginica
113           5.7          2.5           5.0          2.0  Virginica
114           5.8          2.8           5.1          2.4  Virginica
115           6.4          3.2           5.3          2.3  Virginica
116           6.5          3.0           5.5          1.8  Virginica
117           7.7          3.8           6.7          2.2  Virginica
118           7.7          2.6           6.9          2.3  Virginica
119           6.0          2.2           5.0          1.5  Virginica
120           6.9          3.2           5.7          2.3  Virginica
121           5.6          2.8           4.9          2.0  Virginica
122           7.7          2.8           6.7          2.0  Virginica
123           6.3          2.7           4.9          1.8  Virginica
124           6.7          3.3           5.7          2.1  Virginica
125           7.2          3.2           6.0          1.8  Virginica
126           6.2          2.8           4.8          1.8  Virginica
127           6.1          3.0           4.9          1.8  Virginica
128           6.4          2.8           5.6          2.1  Virginica
129           7.2          3.0           5.8          1.6  Virginica
130           7.4          2.8           6.1          1.9  Virginica
131           7.9          3.8           6.4          2.0  Virginica
132           6.4          2.8           5.6          2.2  Virginica
133           6.3          2.8           5.1          1.5  Virginica
134           6.1          2.6           5.6          1.4  Virginica
135           7.7          3.0           6.1          2.3  Virginica
136           6.3          3.4           5.6          2.4  Virginica
137           6.4          3.1           5.5          1.8  Virginica
138           6.0          3.0           4.8          1.8  Virginica
139           6.9          3.1           5.4          2.1  Virginica
140           6.7          3.1           5.6          2.4  Virginica
141           6.9          3.1           5.1          2.3  Virginica
143           6.8          3.2           5.9          2.3  Virginica
144           6.7          3.3           5.7          2.5  Virginica
145           6.7          3.0           5.2          2.3  Virginica
146           6.3          2.5           5.0          1.9  Virginica
147           6.5          3.0           5.2          2.0  Virginica
148           6.2          3.4           5.4          2.3  Virginica
149           5.9          3.0           5.1          1.8  Virginica

We can see all the features of Iris Virginica flower with petal size>1.5.

**Task 3**

In [10]:
print("Minimum\n",iris[iris.variety=='Setosa'].min(),'\n')
print("Maximum\n",iris[iris.variety=='Setosa'].max(),'\n')
print("Average\n",iris[iris.variety=='Setosa'].mean(),'\n')

Minimum
 sepal.length       4.3
sepal.width        2.3
petal.length         1
petal.width        0.1
variety         Setosa
dtype: object 

Maximum
 sepal.length       5.8
sepal.width        4.4
petal.length       1.9
petal.width        0.6
variety         Setosa
dtype: object 

Average
 sepal.length    5.006
sepal.width     3.428
petal.length    1.462
petal.width     0.246
dtype: float64 



In [11]:
print("Minimum\n",iris[iris.variety=='Versicolor'].min(),'\n')
print("Maximum\n",iris[iris.variety=='Versicolor'].max(),'\n')
print("Average\n",iris[iris.variety=='Versicolor'].mean(),'\n')

Minimum
 sepal.length           4.9
sepal.width              2
petal.length             3
petal.width              1
variety         Versicolor
dtype: object 

Maximum
 sepal.length             7
sepal.width            3.4
petal.length           5.1
petal.width            1.8
variety         Versicolor
dtype: object 

Average
 sepal.length    5.936
sepal.width     2.770
petal.length    4.260
petal.width     1.326
dtype: float64 



In [12]:
print("Minimum\n",iris[iris.variety=='Virginica'].min(),'\n')
print("Maximum\n",iris[iris.variety=='Virginica'].max(),'\n')
print("Average\n",iris[iris.variety=='Virginica'].mean(),'\n')

Minimum
 sepal.length          4.9
sepal.width           2.2
petal.length          4.5
petal.width           1.4
variety         Virginica
dtype: object 

Maximum
 sepal.length          7.9
sepal.width           3.8
petal.length          6.9
petal.width           2.5
variety         Virginica
dtype: object 

Average
 sepal.length    6.604082
sepal.width     2.979592
petal.length    5.561224
petal.width     2.028571
dtype: float64 



Hence, we obtained maximum, minimum and average of all the features for all different types of flowers.

# Terrorism

### Preprocessing

In [13]:
terror=pd.read_csv('terrorismData.csv')
terror

Year  Month  Day             Country            State  \
0       1970      7    2  Dominican Republic              NaN   
1       1970      0    0              Mexico          Federal   
2       1970      1    0         Philippines           Tarlac   
3       1970      1    0              Greece           Attica   
4       1970      1    0               Japan          Fukouka   
...      ...    ...  ...                 ...              ...   
181686  2017     12   31             Somalia  Middle Shebelle   
181687  2017     12   31               Syria         Lattakia   
181688  2017     12   31         Philippines      Maguindanao   
181689  2017     12   31               India          Manipur   
181690  2017     12   31         Philippines      Maguindanao   

                             Region           City   Latitude   Longitude  \
0       Central America & Caribbean  Santo Domingo  18.456792  -69.951164   
1                     North America    Mexico city  19.371887  -99.086624   
2                    Southeast Asia        Unknown  15.478598  120.599741   
3                    Western Europe         Athens  37.997490   23.762728   
4                         East Asia        Fukouka  33.580412  130.396361   
...                             ...            ...        ...         ...   
181686           Sub-Saharan Africa  Ceelka Geelow   2.359673   45.385034   
181687   Middle East & North Africa         Jableh  35.407278   35.942679   
181688               Southeast Asia       Kubentog   6.900742  124.437908   
181689                   South Asia         Imphal  24.798346   93.940430   
181690               Southeast Asia  Cotabato City   7.209594  124.241966   

                            AttackType  Killed  Wounded  \
0                        Assassination     1.0      0.0   
1          Hostage Taking (Kidnapping)     0.0      0.0   
2                        Assassination     1.0      0.0   
3                    Bombing/Explosion     NaN      NaN   
4       Facility/Infrastructure Attack     NaN      NaN   
...                                ...     ...      ...   
181686                   Armed Assault     1.0      2.0   
181687               Bombing/Explosion     2.0      7.0   
181688  Facility/Infrastructure Attack     0.0      0.0   
181689               Bombing/Explosion     0.0      0.0   
181690               Bombing/Explosion     0.0      0.0   

                         Target  \
0                  Julio Guzman   
1       Nadine Chaval, daughter   
2                      Employee   
3                  U.S. Embassy   
4                U.S. Consulate   
...                         ...   
181686               Checkpoint   
181687         Hmeymim Air Base   
181688                   Houses   
181689                   Office   
181690                  Unknown   

                                                  Summary  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  12/31/2017: Assailants opened fire on a Somali...   
181687  12/31/2017: Assailants launched mortars at the...   
181688  12/31/2017: Assailants set fire to houses in K...   
181689  12/31/2017: Assailants threw a grenade at a Fo...   
181690  12/31/2017: An explosive device was discovered...   

                                             Group  \
0                                           MANO-D   
1               23rd of September Communist League   
2                                          Unknown   
3                                          Unknown   
4                                          Unknown   
...                                            ...   
181686                                  Al-Shabaab   
181687 

Loaded the dataset.

In [14]:
terror.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Data columns (total 17 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Year         181691 non-null  int64  
 1   Month        181691 non-null  int64  
 2   Day          181691 non-null  int64  
 3   Country      181691 non-null  object 
 4   State        181270 non-null  object 
 5   Region       181691 non-null  object 
 6   City         181257 non-null  object 
 7   Latitude     177135 non-null  float64
 8   Longitude    177134 non-null  float64
 9   AttackType   181691 non-null  object 
 10  Killed       171378 non-null  float64
 11  Wounded      165380 non-null  float64
 12  Target       181055 non-null  object 
 13  Summary      115562 non-null  object 
 14  Group        181691 non-null  object 
 15  Target_type  181691 non-null  object 
 16  Weapon_type  181691 non-null  object 
dtypes: float64(4), int64(3), object(10)
memory usage: 23.6+ MB


We see there are many columns with null values.

In [15]:
terror=terror.fillna({"Target":'Unknown','Summary':'Unknown','Region':'Unknown','City':'Unknown'})

In [16]:
terror=terror.fillna({'Latitude':terror['Latitude'].mean(),'Longitude':terror['Longitude'].mean(),'Killed':terror['Killed'].mean(),'Wounded':terror['Wounded'].mean()})

In [17]:
terror.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Data columns (total 17 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Year         181691 non-null  int64  
 1   Month        181691 non-null  int64  
 2   Day          181691 non-null  int64  
 3   Country      181691 non-null  object 
 4   State        181270 non-null  object 
 5   Region       181691 non-null  object 
 6   City         181691 non-null  object 
 7   Latitude     181691 non-null  float64
 8   Longitude    181691 non-null  float64
 9   AttackType   181691 non-null  object 
 10  Killed       181691 non-null  float64
 11  Wounded      181691 non-null  float64
 12  Target       181691 non-null  object 
 13  Summary      181691 non-null  object 
 14  Group        181691 non-null  object 
 15  Target_type  181691 non-null  object 
 16  Weapon_type  181691 non-null  object 
dtypes: float64(4), int64(3), object(10)
memory usage: 23.6+ MB


Filled the NA values.

In [18]:
terror[terror.duplicated()==True]

Year  Month  Day        Country         State  \
68      1970      2   16  United States    California   
88      1970      2   23  United States      Nebraska   
89      1970      2   23  United States      Nebraska   
90      1970      2   23  United States      Nebraska   
337     1970      6    9         Jordan         Amman   
...      ...    ...  ...            ...           ...   
181397  2017     12   18       Pakistan   Balochistan   
181431  2017     12   20    Philippines   Maguindanao   
181587  2017     12   27          India  Chhattisgarh   
181634  2017     12   29         Israel      Southern   
181657  2017     12   30          Kenya       Garissa   

                            Region              City   Latitude   Longitude  \
68                   North America          Berkeley  37.874043 -122.280022   
88                   North America  South Sioux City  42.470310  -96.413949   
89                   North America  South Sioux City  42.470310  -96.413949   
90                   North America  South Sioux City  42.470310  -96.413949   
337     Middle East & North Africa             Amman  31.950001   35.933331   
...                            ...               ...        ...         ...   
181397                  South Asia           Mangoli  28.466810   68.274078   
181431              Southeast Asia         Tamontaka   7.172751  124.229664   
181587                  South Asia          Basaguda  18.561113   80.920395   
181634  Middle East & North Africa           Unknown  31.506981   34.718904   
181657          Sub-Saharan Africa             Ijara  -1.595245   40.514259   

                                 AttackType  Killed  Wounded  \
68                        Bombing/Explosion     0.0      0.0   
88                        Bombing/Explosion     0.0      0.0   
89                        Bombing/Explosion     0.0      0.0   
90                        Bombing/Explosion     0.0      0.0   
337     Hostage Taking (Barricade Incident)     0.0      0.0   
...                                     ...     ...      ...   
181397                    Bombing/Explosion     0.0      0.0   
181431                    Bombing/Explosion     0.0      0.0   
181587                    Bombing/Explosion     0.0      0.0   
181634                    Bombing/Explosion     0.0      0.0   
181657                        Armed Assault     0.0      0.0   

                                                   Target  \
68                                       department store   
88      Power Transmission Pole in the Dakota City/Sio...   
89      Power Transmission Pole in the Dakota City/Sio...   
90      Power Transmission Pole in the Dakota City/Sio...   
337                                                Hotels   
...                                                   ...   
181397                                  Electricity Tower   
181431                                         Checkpoint   
181587                                            Unknown   
181634                                            Unknown   
181657                                     Police Station   

                                                  Summary  \
68                                                Unknown   
88      2/23/1970: In a series of related attacks, unk...   
89      2/23/1970: In a series of related attacks, unk...   
90      2/23/1970: In a series of related attacks, unk...   
337                                               Unknown   
...                                                   ...   
181397  12/18/2017: An explosive device detonated near...   
181431  12/20/2017: Assailants threw a grenade at a mi...   
181587  12/00/2017: Sometime between December 24, 2017...   
181634  12/29/2017: Assailants in Gaza fired a rocket ...   
181657  12/30/2017: Assailants raided a police station...   

                                                    Group Target_type  \
68                                                Unknown    Busin

Many columns are duplicated.

In [19]:
terror=terror.drop_duplicates()

In [20]:
terror=terror.reset_index()

In [21]:
terror=terror.drop(columns=['index'])

Dropped the duplicates and resetted the indexing.

In [22]:
for i in range(len(terror.Month)):
    if terror.Month[i]==0:
        terror.Month[i]=1
for i in range(len(terror.Day)):
    if terror.Day[i]==0:
        terror.Day[i]=1

C:\Users\Aviral.HP-PAVILION\anaconda3\envs\Deep learning\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Users\Aviral.HP-PAVILION\anaconda3\envs\Deep learning\lib\site-packages\ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


Many entries in 'Month' and 'Day' were observed to be 0 which isn't correct and hence we change it to 1.

In [23]:
terror["Date"]=pd.to_datetime(terror.iloc[:,:3],infer_datetime_format=True)

In [24]:
terror=terror.drop(columns=['Year','Month','Day'])

Made a column with Dates of the attack of 'datetime' type.

In [25]:
terror

Country            State                       Region  \
0       Dominican Republic              NaN  Central America & Caribbean   
1                   Mexico          Federal                North America   
2              Philippines           Tarlac               Southeast Asia   
3                   Greece           Attica               Western Europe   
4                    Japan          Fukouka                    East Asia   
...                    ...              ...                          ...   
172182             Somalia  Middle Shebelle           Sub-Saharan Africa   
172183               Syria         Lattakia   Middle East & North Africa   
172184         Philippines      Maguindanao               Southeast Asia   
172185               India          Manipur                   South Asia   
172186         Philippines      Maguindanao               Southeast Asia   

                 City   Latitude   Longitude                      AttackType  \
0       Santo Domingo  18.456792  -69.951164                   Assassination   
1         Mexico city  19.371887  -99.086624     Hostage Taking (Kidnapping)   
2             Unknown  15.478598  120.599741                   Assassination   
3              Athens  37.997490   23.762728               Bombing/Explosion   
4             Fukouka  33.580412  130.396361  Facility/Infrastructure Attack   
...               ...        ...         ...                             ...   
172182  Ceelka Geelow   2.359673   45.385034                   Armed Assault   
172183         Jableh  35.407278   35.942679               Bombing/Explosion   
172184       Kubentog   6.900742  124.437908  Facility/Infrastructure Attack   
172185         Imphal  24.798346   93.940430               Bombing/Explosion   
172186  Cotabato City   7.209594  124.241966               Bombing/Explosion   

          Killed   Wounded                   Target  \
0       1.000000  0.000000             Julio Guzman   
1       0.000000  0.000000  Nadine Chaval, daughter   
2       1.000000  0.000000                 Employee   
3       2.403272  3.167668             U.S. Embassy   
4       2.403272  3.167668           U.S. Consulate   
...          ...       ...                      ...   
172182  1.000000  2.000000               Checkpoint   
172183  2.000000  7.000000         Hmeymim Air Base   
172184  0.000000  0.000000                   Houses   
172185  0.000000  0.000000                   Office   
172186  0.000000  0.000000                  Unknown   

                                                  Summary  \
0                                                 Unknown   
1                                                 Unknown   
2                                                 Unknown   
3                                                 Unknown   
4                                                 Unknown   
...                                                   ...   
172182  12/31/2017: Assailants opened fire on a Somali...   
172183  12/31/2017: Assailants launched mortars at the...   
172184  12/31/2017: Assailants set fire to houses in K...   
172185  12/31/2017: Assailants threw a grenade at a Fo...   
172186  12/31/2017: An explosive device was discovered...   

                                             Group  \
0                                           MANO-D   
1               23rd of September Communist League   
2                                          Unknown   
3                                          Unknown   
4                                          Unknown   
...                                            ...   
172182                                  Al-Shabaab   
172183                           Muslim extremists   
172184  Bangsamoro Islamic Freedom Movement (BIFM)   
172185                                     Unknown   
172186                                     Unknown   

                        Target_type Weapon_type       Date  
0       Private Citizens & Property     U

### Tasks

**Task 1**

In [26]:
terror.Country.value_counts().head()

Iraq           23458
Pakistan       13795
Afghanistan    12512
India          11585
Colombia        7504
Name: Country, dtype: int64

We see that Iraq had the most terror attacks.

In [27]:
terror_iraq=terror[terror.Country=='Iraq']

In [28]:
pd.DatetimeIndex(terror_iraq['Date']).year.value_counts().head()

2014    3651
2016    2850
2013    2771
2015    2671
2017    2292
Name: Date, dtype: int64

We can infer that most attacks in Iraq were held in 2014.

**Task 2**

In [29]:
terror_india=terror[terror.Country=="India"]
govt=terror_india[terror_india.Date>='2014-05-26']

In [30]:
govt

Country              State      Region           City   Latitude  \
124685   India     Andhra Pradesh  South Asia    Bandlapalli  14.714739   
124713   India        West Bengal  South Asia   Sandeshkhali  22.356709   
124719   India        West Bengal  South Asia   Sandeshkhali  22.363882   
124742   India  Jammu and Kashmir  South Asia          Khour  32.601778   
124770   India             Odisha  South Asia      Bikrampur  22.338542   
...        ...                ...         ...            ...        ...   
172159   India             Kerala  South Asia  Koothuparamba  11.831902   
172161   India       Chhattisgarh  South Asia       Toylanka  18.802725   
172168   India  Jammu and Kashmir  South Asia      Lethapora  33.966527   
172180   India              Assam  South Asia        Hungrum  25.180162   
172185   India            Manipur  South Asia         Imphal  24.798346   

        Longitude                      AttackType  Killed  Wounded  \
124685  77.810281               Bombing/Explosion     0.0      0.0   
124713  88.881991                   Armed Assault     0.0     12.0   
124719  88.878679                   Armed Assault     0.0     11.0   
124742  74.809414               Bombing/Explosion     0.0      0.0   
124770  86.178146                   Assassination     1.0      0.0   
...           ...                             ...     ...      ...   
172159  75.565432               Bombing/Explosion     0.0      0.0   
172161  81.497666  Facility/Infrastructure Attack     0.0      0.0   
172168  74.964225                   Armed Assault     8.0      3.0   
172180  93.015788     Hostage Taking (Kidnapping)     0.0      0.0   
172185  93.940430               Bombing/Explosion     0.0      0.0   

                                                   Target  \
124685                                    Party Supporter   
124713                                      Party Workers   
124719                                            Members   
124742                                            Unknown   
124770                    Journalist: Tarun Kumar Acharya   
...                                                   ...   
172159                       Koothuparamba Police Station   
172161                             Road Construction Site   
172168                                               Camp   
172180  Personal Security Officer of Council Member Ih...   
172185                                             Office   

                                                  Summary  \
124685  05/26/2014: An explosive device was discovered...   
124713  05/26/2014: Assailants opened fire on Bharatiy...   
124719  05/26/2014: Assailants opened fire and threw e...   
124742  05/27/2014: An explosive device was discovered...   
124770  05/27/2014: Assailants stabbed and killed Taru...   
...                                                   ...   
172159  12/30/2017: Assailants threw an explosive devi...   
172161  12/30/2017: Assailants set fire to seven vehic...   
172168  12/31/2017: Assailants armed with grenades and...   
172180  12/31/2017: Assailants abducted Prafulla Phuka...   
172185  12/31/2017: Assailants threw a grenade at a Fo...   

                                                 Group  \
124685                                         Unknown   
124713                                         Unknown   
124719                                         Unknown   
124742                                         Unknown   
124770                                         Unknown   
...                                                ...   
172159                                         Unknown   
172161  Communist Party of India - Maoist (CPI-Maoist)   
172168                          Jaish-e-Mohammad (JeM)   
172180                        Zeliangrong United Front   
172185                                         Unknown   

                        Target_type Weapon_type       Date  
124685  Private Citizens & Property  Expl

In [31]:
govt.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3307 entries, 124685 to 172185
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   Country      3307 non-null   object        
 1   State        3307 non-null   object        
 2   Region       3307 non-null   object        
 3   City         3307 non-null   object        
 4   Latitude     3307 non-null   float64       
 5   Longitude    3307 non-null   float64       
 6   AttackType   3307 non-null   object        
 7   Killed       3307 non-null   float64       
 8   Wounded      3307 non-null   float64       
 9   Target       3307 non-null   object        
 10  Summary      3307 non-null   object        
 11  Group        3307 non-null   object        
 12  Target_type  3307 non-null   object        
 13  Weapon_type  3307 non-null   object        
 14  Date         3307 non-null   datetime64[ns]
dtypes: datetime64[ns](1), float64(4), object(10)
mem

There are 3307 entries in this dataframe which is the number of attacks done in this period.

In [32]:
govt.Group.value_counts().head()

Unknown                                           1271
Maoists                                            843
Communist Party of India - Maoist (CPI-Maoist)     323
Hizbul Mujahideen (HM)                              78
Lashkar-e-Taiba (LeT)                               69
Name: Group, dtype: int64

Ignoring the Unknown attacks, Maoists were the most active group during this period with 843 attacks.